src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -0,0 +1,3768 @@
+            valid_ratios=valid_ratios,
+            self_attn_mask=None,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        if not return_dict:
+            enc_outputs = tuple(value for value in [enc_outputs_class, enc_outputs_coord_logits] if value is not None)
+            tuple_outputs = (init_reference_points,) + decoder_outputs + encoder_outputs + enc_outputs
+
+            return tuple_outputs
+
+        return GroundingDINOModelOutput(
+            init_reference_points=init_reference_points,
+            last_hidden_state=decoder_outputs.last_hidden_state,
+            intermediate_hidden_states=decoder_outputs.intermediate_hidden_states,
+            intermediate_reference_points=decoder_outputs.intermediate_reference_points,
+            decoder_hidden_states=decoder_outputs.hidden_states,
+            decoder_attentions=decoder_outputs.attentions,
+            decoder_cross_attentions_vision=decoder_outputs.vision_cross_attentions,
+            decoder_cross_attentions_text=decoder_outputs.text_cross_attentions,
+            encoder_last_hidden_state_vision=encoder_outputs.last_hidden_state_vision,
+            encoder_last_hidden_state_text=encoder_outputs.last_hidden_state_text,
+            encoder_hidden_states_vision=encoder_outputs.hidden_states_vision,
+            encoder_hidden_states_text=encoder_outputs.hidden_states_text,
+            encoder_attentions_vision=encoder_outputs.attentions_vision,
+            encoder_attentions_text=encoder_outputs.attentions_text,
+            encoder_cross_attentions_vision=encoder_outputs.cross_attentions_vision,
+            encoder_cross_attentions_text=encoder_outputs.cross_attentions_text,
+            enc_outputs_class=enc_outputs_class,
+            enc_outputs_coord_logits=enc_outputs_coord_logits,
+        )
+
+
+@add_start_docstrings(
+    """
+    Grounding DINO Model (consisting of a backbone and encoder-decoder Transformer) with object detection heads on top,
+    for tasks such as COCO detection.
+    """,
+    GROUNDING_DINO_START_DOCSTRING,
+)
+class GroundingDINOForObjectDetection(GroundingDINOPreTrainedModel):
+    # When using clones, all layers > 0 will be clones, but layer 0 *is* required
+    _tied_weights_keys = [r"bbox_embed\.[1-9]\d*", r"class_embed\.[1-9]\d*"]
+
+    def __init__(self, config: GroundingDINOConfig):
+        super().__init__(config)
+
+        # Deformable DETR encoder-decoder model
+        self.model = GroundingDINOModel(config)
+
+        # Detection heads on top
+        _class_embed = GroundingDINOContrastiveEmbedding(config)
+        _bbox_embed = GroundingDINOMLPPredictionHead(
+            input_dim=config.d_model, hidden_dim=config.d_model, output_dim=4, num_layers=3
+        )
+
+        nn.init.constant_(_bbox_embed.layers[-1].weight.data, 0)
+        nn.init.constant_(_bbox_embed.layers[-1].bias.data, 0)
```

Comment:
we should move things like these (actually all weight initializations, methods like `reset_parameters`) to the `_init_weights` method of GroundingDINOPreTrained model

Reply:
Sure, I left those just because I modified this class based on `DeformableDetrForObjectDetection` which leaves these inits there as well